lib/seahorn/HornifyModule.cc

```diff
@@ -142,6 +169,19 @@ bool HornifyModule::runOnModule(Module &M) {
   if (Step == hm_detail::CLP_SMALL_STEP ||
       Step == hm_detail::CLP_FLAT_SMALL_STEP)
     m_sem.reset(new ClpOpSem(m_efac, *this, M.getDataLayout(), TL));
+  else if (InterProcMem) {
+    ShadowMemPass * smp = getAnalysisIfAvailable<sea_dsa::ShadowMemPass>();
+    assert(smp);
+    ShadowMem &shadowmem_analysis =smp->getShadowMem();
+
+    CompleteCallGraph *ccg = getAnalysisIfAvailable<sea_dsa::CompleteCallGraph>();
+    std::shared_ptr<InterMemPreProc> preproc(new InterMemPreProc(ccg, shadowmem_analysis));
+
+    preproc->runOnModule(M);
+
+    m_sem.reset(new UfoOpMemSem(m_efac, *this, M.getDataLayout(), preproc, TL,
+                                abs_fns, &shadowmem_analysis));
```

Comment:
A better design would be to add setters on `UfoOpMemSem` to communicate extra analyses that it needs.
Then, `UfoOpMemSem` works like `UfoOpSem` when the extra fields are not set, and like itself when they are.
This will also eliminate the need for two versions of `UfoOpSem`

Reply:
I'm working on this